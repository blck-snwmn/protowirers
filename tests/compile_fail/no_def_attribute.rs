use protowirers::*;

#[derive(Proto)]
struct Sample {
    #[defx(field_num = 1, def_type = "int32")]
    s: u32,
}

fn main() {}
