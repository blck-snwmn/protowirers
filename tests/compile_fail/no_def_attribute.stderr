error: #[def(...)] attribute requires.
 --> $DIR/no_def_attribute.rs:6:5
  |
6 |     s: u32,
  |     ^

error: cannot find attribute `defx` in this scope
 --> $DIR/no_def_attribute.rs:5:7
  |
5 |     #[defx(field_num = 1, def_type = "int32")]
  |       ^^^^ help: a derive helper attribute with a similar name exists: `def`
